### PR TITLE
Fixes #23624 "nim check crash"

### LINKED ISSUE #23624: nim check crashes
### Description

On Windows, anything that calls `allocShared0`  fails subsequent checks with `nim check`

```nim

proc myFunc(a: string) =
  var x: int = ""

import std/os
static:
  echo getCurrentDir()

proc myFunc2(a: string) =  
  var x: int = ""  #this check not marked as error
```

### Nim Version

Nim Compiler Version 2.1.1 [Windows: amd64]
Compiled at 2024-03-06
Copyright (c) 2006-2024 by Andreas Rumpf

### Current Output

_No response_

### Expected Output

_No response_

### Possible Solution

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by jmgomez**

### CODE REVIEW

**Comment by Araq** [comment]
Too arbitrary for my taste.

**Comment by jmgomez** [comment]
@Araq I know it's arbitrary but the issue is that it reaches a global error on purpose. It seems like the compiler cant recover from global errors, hence it halts. That's why the workaround was to just take `nimcheck` into consideration.

File: compiler/vmgen.nim [hunk 1]
```diff
@@ -1524,8 +1525,12 @@ proc setSlot(c: PCtx; v: PSym) =
     v.position = getFreeRegister(c, if v.kind == skLet: slotFixedLet else: slotFixedVar, start = 1)
 
 proc cannotEval(c: PCtx; n: PNode) {.noinline.} =
-  globalError(c.config, n.info, "cannot evaluate at compile time: " &
+  if c.config.cmd == cmdCheck:
+    localError(c.config, n.info, "cannot evaluate at compile time: " &
```
Comment (Araq) [inline compiler/vmgen.nim:1529]: Why? It's very dangerous to do this. You should at least turn `cannotEval` into a `template` then and use `return` after the `localError`.

**Comment by Araq** [comment]
Changed my mind, at least it's better than before.

**Pull request merged**